coordinator/tributary/src/blockchain.rs

```diff
@@ -28,39 +29,21 @@ pub(crate) struct Blockchain<D: Db, T: TransactionTrait> {
   pub(crate) next_block_notifications: VecDeque<tokio::sync::oneshot::Sender<()>>,
 }
 
-impl<D: Db, T: TransactionTrait> Blockchain<D, T> {
-  fn tip_key(genesis: [u8; 32]) -> Vec<u8> {
-    D::key(b"tributary_blockchain", b"tip", genesis)
-  }
-  fn block_number_key(&self) -> Vec<u8> {
-    D::key(b"tributary_blockchain", b"block_number", self.genesis)
-  }
-  fn block_key(genesis: &[u8], hash: &[u8; 32]) -> Vec<u8> {
-    D::key(b"tributary_blockchain", b"block", [genesis, hash].concat())
-  }
-  fn block_hash_key(genesis: &[u8], block_number: u32) -> Vec<u8> {
-    D::key(b"tributary_blockchain", b"block_hash", [genesis, &block_number.to_le_bytes()].concat())
-  }
-  fn commit_key(genesis: &[u8], hash: &[u8; 32]) -> Vec<u8> {
-    D::key(b"tributary_blockchain", b"commit", [genesis, hash].concat())
-  }
-  fn block_after_key(genesis: &[u8], hash: &[u8; 32]) -> Vec<u8> {
-    D::key(b"tributary_blockchain", b"block_after", [genesis, hash].concat())
-  }
-  fn unsigned_included_key(genesis: &[u8], hash: &[u8; 32]) -> Vec<u8> {
-    D::key(b"tributary_blockchain", b"unsigned_included", [genesis, hash].concat())
-  }
-  fn provided_included_key(genesis: &[u8], hash: &[u8; 32]) -> Vec<u8> {
-    D::key(b"tributary_blockchain", b"provided_included", [genesis, hash].concat())
-  }
-  fn next_nonce_key(&self, signer: &<Ristretto as Ciphersuite>::G) -> Vec<u8> {
-    D::key(
-      b"tributary_blockchain",
-      b"next_nonce",
-      [self.genesis.as_ref(), signer.to_bytes().as_ref()].concat(),
-    )
+create_db!(
+  TributaryBlockchainDb {
+    TipsDb: (genesis: [u8; 32]) -> [u8; 32],
+    BlockNumberDb: (genesis: [u8; 32]) -> Vec<u8>,
+    BlockDb: (genesis:  [u8; 32], hash: &[u8; 32]) -> Vec<u8>,
+    BlockHashDb: (genesis: [u8; 32], block_number: u32) -> [u8; 32],
+    CommitDb: (genesis: [u8; 32], block: &[u8; 32]) -> Vec<u8>,
+    BlockAfterDb: (genesis: [u8; 32], hash: [u8; 32]) -> [u8; 32],
+    UnsignedIncludedDb: (genesis: [u8; 32], hash: [u8; 32]) -> Vec<u8>,
+    ProvidedIncludedDb: (genesis: [u8; 32], hash: [u8; 32]) -> [u8; 0],
+    NextNonceDb: (genesis: [u8; 32], hash: [u8; 32]) -> Vec<u8>
```

Comment:
NextNonce is by `signer: [u8; 32]`, not `hash: [u8; 32]`. I also believe this `Vec<u8>`  should be a `u32`